espresso/streamer.go

```diff
@@ -188,21 +231,19 @@ func (s *EspressoStreamer) Start(ctx context.Context) error {
 			return fmt.Errorf("timeout while queueing messages from hotshot")
 		}
 	}
-
-	return nil
 }
 
 // TODO this logic might be slightly different between batcher and derivation
-func (s *EspressoStreamer) Next(ctx context.Context) *EspressoBatchI {
+func (s *EspressoStreamer[B]) Next(ctx context.Context) *B {
+	if s.BatchBuffer.Len() > 0 && (*s.BatchBuffer.Peek()).Number() != s.BatchPos {
+		log.Warn("Batch buffer is being weird", "batchPos", s.BatchPos, "batchBuffer", s.BatchBuffer)
```

Comment:
Extraneous logging, nothing weird about it. Added when I was debugging with strictly sequential batch posting.

Reply:
Removed in https://github.com/EspressoSystems/optimism-espresso-integration/pull/96/commits/99e32de72a7e322476c2183523d45f47cf0d014c.